react-components/src/components/RuleBasedOutputs/utils.ts

```diff
@@ -64,61 +82,100 @@ const checkStringExpressionStatement = (
 
   return expressionResult;
 };
+
+const getTriggerNumericData = (
+  triggerTypeData: TriggerTypeData[],
+  trigger: MetadataRuleTrigger | TimeseriesRuleTrigger
+): number | undefined => {
+  const currentTriggerData = triggerTypeData.find(
+    (triggerType) => triggerType.type === trigger?.type
+  );
+
+  if (currentTriggerData === undefined) return;
+
+  const dataTrigger =
+    currentTriggerData.type === 'metadata' && trigger.type === 'metadata'
+      ? Number(currentTriggerData.asset[trigger.type]?.[trigger.key])
+      : currentTriggerData.type === 'timeseries' && trigger.type === 'timeseries'
+        ? getTriggerTimeseriesNumericData(currentTriggerData, trigger)
+        : undefined;
+
+  return dataTrigger;
+};
+
+const getTriggerTimeseriesNumericData = (
+  triggerTypeData: TriggerTypeData,
+  trigger: TimeseriesRuleTrigger
+): number | undefined => {
+  if (triggerTypeData.type === 'timeseries' && trigger.type === 'timeseries') {
+    const timeseriesWithDatapoints = triggerTypeData.timeseries.timeseriesWithDatapoints;
+
+    const dataFound = timeseriesWithDatapoints.find(
+      (item) => item.externalId === trigger.externalId
+    );
+
+    const datapoint = dataFound?.datapoints[dataFound?.datapoints.length - 1]?.value;
+
+    return Number(datapoint);
+  }
+  return undefined;
```

Comment:
Can we do early return in this function?

Reply:
https://github.com/cognitedata/reveal/pull/4414/commits/b343e596937f0a83e23dbf4967a1762fd4e7a87c